src/components/Main.js

```diff
@@ -0,0 +1,105 @@
+import {nanoid} from 'nanoid';
+import {useState, useEffect} from 'react';
+
+import data from '../data.json';
+import StyledMain from '../style/StyledMain';
+
+export default function Form() {
+  const [selectedBodypart, setSelectedBodypart] = useState('');
+  const [plans, setPlans] = useState([]);
+  const [weeks, setWeeks] = useState(JSON.parse(localStorage.getItem('my_plan')) || []);
+  //handlesubmit
+  function handleSubmit(event) {
+    event.preventDefault();
+    saveWeek();
+  }
+  //setting the day workout
+  function createPlan(event) {
+    const newPlan = event.target.value;
+    setPlans([...plans, {name: newPlan, id: nanoid()}]);
+  }
+  // workouts store
+  useEffect(() => {
+    localStorage.setItem('my_plan', JSON.stringify(weeks));
+  }, [weeks]);
+  //setting the sets
+  function updateSets(id, value) {
+    setPlans([...plans.slice(0, id), {...plans[id], sets: value}, ...plans.slice(id + 1)]);
+  }
+  //save multiple workouts
+  function saveWeek() {
+    const newSet = {exercise: plans, bodyPart: selectedBodypart};
+    setWeeks([...weeks, newSet]);
+    setPlans([]);
+    setSelectedBodypart('');
+  }
+  //delete day-plan
+  function deletMyPlan(id) {
+    setWeeks([...weeks.slice(0, id), ...weeks.slice(id + 1)]);
+  }
```

Comment:
Please rename this to `index` as this is an `index`:
```suggestion
  function deletMyPlan(index) {
    setWeeks([...weeks.slice(0, index), ...weeks.slice(index + 1)]);
  }
```